File: Guia4-DSM-BF180436/Guia4_Ejercicio1/app/src/main/java/sv/edu/udb/guia4_ejercicio1/LoginActivity.java
package sv.edu.udb.guia4_ejercicio1;

 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 import android.content.Intent;
 import android.os.Bundle;
 import android.text.TextUtils;
 import android.view.View;
 import android.widget.Button;
 import android.widget.EditText;
 import android.widget.ProgressBar;
 import android.widget.Toast;
 import com.google.android.gms.auth.api.signin.GoogleSignIn;
 import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
 import com.google.android.gms.auth.api.signin.GoogleSignInClient;
 import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
 import com.google.android.gms.common.api.ApiException;
 import com.google.android.gms.tasks.OnCompleteListener;
 import com.google.android.gms.tasks.Task;
 import com.google.firebase.auth.AuthCredential;
 import com.google.firebase.auth.AuthResult;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.GoogleAuthProvider;

 import java.lang.reflect.Array;
 import java.util.Arrays;

public class LoginActivity extends AppCompatActivity {

    private EditText emailTV, passwordTV;
    private Button loginBtn;
    private ProgressBar progressBar;
    private FirebaseAuth mAuth;
    Button registerBtn, facebookBtn, googleBtn;
    private GoogleSignInClient mGoogleSignInClient;
    private final static int RC_SIGN_IN = 123;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        createRequest();

        mAuth = FirebaseAuth.getInstance();

        initializeUI();


        registerBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(LoginActivity.this, RegistrationActivity.class);
                startActivity(intent);
            }
        });

        loginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginUserAccount();
            }
        });

        googleBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginGoogle();
            }
        });

        facebookBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginFacebook();
            }
        });
    }

    private void createRequest() {
        // Configure Google Sign In
        GoogleSignInOptions gso = new GoogleSignInOptions
                .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                .requestIdToken("543261323216-r9uieg4alni25ltmh8i5ifhpfacl94d8.apps.googleusercontent.com")
                .requestEmail()
                .build();

        // Build a GoogleSignInClient with the options specified by gso.
        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
    }

    private void signIn() {
        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
        startActivityForResult(signInIntent, RC_SIGN_IN);
    }

    private void loginUserAccount() {
        progressBar.setVisibility(View.VISIBLE);

        String email, password;
        email = emailTV.getText().toString();
        password = passwordTV.getText().toString();

        if (TextUtils.isEmpty(email)) {
            Toast.makeText(getApplicationContext(), "Please enter email...", Toast.LENGTH_LONG).show();
            return;
        }
        if (TextUtils.isEmpty(password)) {
            Toast.makeText(getApplicationContext(), "Please enter password!", Toast.LENGTH_LONG).show();
            return;
        }

        mAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            Toast.makeText(getApplicationContext(), "Login successful!", Toast.LENGTH_LONG).show();
                            Intent intent = new Intent(LoginActivity.this, DashboardActivity.class);
                            startActivity(intent);
                            progressBar.setVisibility(View.GONE);
                        }
                        else {
                            Toast.makeText(getApplicationContext(), "Login failed! Please try again later", Toast.LENGTH_LONG).show();
                            progressBar.setVisibility(View.GONE);
                        }
                    }
                });
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
        if (requestCode == RC_SIGN_IN) {
            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
            try {
                // Google Sign In was successful, authenticate with Firebase
                GoogleSignInAccount account = task.getResult(ApiException.class);
                firebaseAuthWithGoogle(account.getIdToken());
            } catch (ApiException e) {
                Toast.makeText(getApplicationContext(), "Error -> " + e.getMessage(), Toast.LENGTH_LONG).show();
                progressBar.setVisibility(View.GONE);
            }
        }
    }

    private void firebaseAuthWithGoogle(String idToken) {
        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
        mAuth.signInWithCredential(credential)
                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            // Sign in success, update UI with the signed-in user's information
                            Toast.makeText(getApplicationContext(), "Login successful!", Toast.LENGTH_LONG).show();
                            Intent intent = new Intent(LoginActivity.this, DashboardActivity.class);
                            startActivity(intent);
                            progressBar.setVisibility(View.GONE);
                        } else {
                            // If sign in fails, display a message to the user.
                            Toast.makeText(getApplicationContext(), "Login failed! Please try again later", Toast.LENGTH_LONG).show();
                            progressBar.setVisibility(View.GONE);
                        }
                    }
                });
    }

    private void loginGoogle() {
        progressBar.setVisibility(View.VISIBLE);
        signIn();
    }

    private void loginFacebook() {
        Intent intent = new Intent(LoginActivity.this, AuthFacebook.class);
        startActivity(intent);
    }

    private void initializeUI() {
        emailTV = findViewById(R.id.email);
        passwordTV = findViewById(R.id.password);

        loginBtn = findViewById(R.id.login);
        progressBar = findViewById(R.id.progressBar);

        registerBtn = findViewById(R.id.registerbtn);

        facebookBtn = findViewById(R.id.facebookbtn);
        googleBtn = findViewById(R.id.googlebtn);
    }
}

File: Desafio1-DSM-BF180436/app/src/main/java/sv/edu/udb/desafio1_bf180436/loginActivity.kt
package sv.edu.udb.desafio1_bf180436

import android.content.ContentValues
import android.content.Intent
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import java.lang.Exception

class loginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        val buttonlog: Button = findViewById(R.id.btn_login)
        val txtUser: EditText = findViewById(R.id.txtuser)
        val txtPassword: EditText = findViewById(R.id.txtpassword)
        buttonlog.setOnClickListener {
            try {
                val admin: AdminSQLiteHelper = AdminSQLiteHelper(this, "administracion", null, 2)
                val bd: SQLiteDatabase = admin.writableDatabase
                var user = txtUser.text.toString()
                var password = txtPassword.text.toString()
                var fila: Cursor = bd.rawQuery("SELECT password FROM usuarios WHERE user = '" + user + "'", null)
                if (fila.moveToFirst()) {
                    if (password == fila.getString(0)) {
                        txtUser.setText("")
                        txtPassword.setText("")
                        val intent: Intent = Intent(this, MainActivity::class.java)
                        startActivity(intent)
                    }
                    else {
                        Toast.makeText(this, "Usuario/Contraseña incorrecta", Toast.LENGTH_SHORT).show()
                    }
                }
                else {
                    Toast.makeText(this, "El usuario ingresado no existe", Toast.LENGTH_SHORT).show()
                }
            }
            catch (e: Exception) {
                Log.d("Error", e.toString())
            }
        }
        val buttonew: Button = findViewById(R.id.btn_login_new)
        buttonew.setOnClickListener {
            try {
                var exits: Boolean = false
                val admin: AdminSQLiteHelper = AdminSQLiteHelper(this, "administracion", null, 2)
                val bd: SQLiteDatabase = admin.writableDatabase
                var user = txtUser.text.toString()
                var password = txtPassword.text.toString()
                var registro: ContentValues = ContentValues()
                registro.put("user", user)
                registro.put("password", password)
                bd.insertOrThrow("usuarios", null, registro)
                bd.close()
                exits = true
                if (exits) {
                    txtUser.setText("")
                    txtPassword.setText("")
                    val intent: Intent = Intent(this, MainActivity::class.java)
                    startActivity(intent)
                }
                else {
                    Toast.makeText(this, "No se ha podido crear el usuario.", Toast.LENGTH_SHORT).show()
                }
            }
            catch (e: Exception) {
                Log.d("Error", e.toString())
            }
        }
    }
}